providers/zenvia/src/lib/zenvia.provider.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+  SmsEventStatusEnum,
+  ISMSEventBody,
+} from '@novu/stateless';
+import { ZenviaParams } from '../types/param';
+import axios from 'axios';
+
+export class ZenviaProvider implements ISmsProvider {
+  id = 'zenvia';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  private axiosInstance = axios.create();
+
+  constructor(
+    private config: {
+      apiKey: string;
+    }
+  ) {}
+
+  async sendMessage(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    let response = {};
+
+    const contentParse = JSON.parse(options.content);
+
+    switch (contentParse.type_provider) {
+      case 'SMS':
+        response = this.sendSMS(options);
+        break;
+      case 'WHATSAPP':
```

Comment:
To be honest for me it doesn't feel right - it kinda breaks responsibility of SMS channel.

Reply:
So should I put WhatsApp sending as a chat category?